### PR TITLE
sys/byteorder: add helpers to convert from little endian to host byte order

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

This is usually a NOP, but a helper functions to allow for writing code that would also work on mythical Big Endian machines.

### CODE REVIEW

**Comment by aabadie** [comment]
Seems to be the third PR trying to address this, after #10986 (initial one) and #14508.

**Comment by benpicco** [comment]
Oh, didn't notice. Well, let's hope third time's a charm then :smile: 
I added the extension to the test from #10986

**Comment by kaspar030** [comment]
I think one commit has a typo (byteoder -> byteorder).

**Comment by benpicco** [comment]
Now the spelling should be in order.

File: tests/unittests/tests-core/tests-core-byteorder.c [hunk 1]
```diff
@@ -136,6 +160,9 @@ Test *tests_core_byteorder_tests(void)
         new_TestFixture(test_byteorder_htobebufs),
         new_TestFixture(test_byteorder_bebuftohl),
         new_TestFixture(test_byteorder_htobebufl),
+        new_TestFixture(test_byteorder_host_to_le_16),
+        new_TestFixture(test_byteorder_host_to_le_32),
+        new_TestFixture(test_byteorder_host_to_le_64),
```
Comment (aabadie) [inline tests/unittests/tests-core/tests-core-byteorder.c:165]:
```suggestion
        new_TestFixture(test_byteorder_htols),
        new_TestFixture(test_byteorder_htoll),
        new_TestFixture(test_byteorder_htolll),
```

File: tests/unittests/tests-ieee802154/tests-ieee802154.c [hunk 2]
```diff
@@ -23,11 +23,6 @@
 #include "unittests-constants.h"
 #include "tests-ieee802154.h"
 
-static inline le_uint16_t byteorder_htols(uint16_t v)
-{
-    return byteorder_btols(byteorder_htons(v));
-}
-
```
Comment (aabadie) [inline tests/unittests/tests-ieee802154/tests-ieee802154.c:30]: Is this related ?

File: tests/unittests/tests-core/tests-core-byteorder.c [hunk 3]
```diff
@@ -120,6 +120,30 @@ static void test_byteorder_htobebufl(void)
...
+
+static void test_byteorder_host_to_le_32(void)
+{
+    static const uint32_t host = 0x12345678ul;
+    le_uint32_t le = { .u8 = { 0x78, 0x56, 0x34, 0x12, } };
+    TEST_ASSERT_EQUAL_INT(le.u32, byteorder_htoll(host).u32);
+    TEST_ASSERT_EQUAL_INT(host, byteorder_ltohl(le));
+}
+
+static void test_byteorder_host_to_le_64(void)
```
Comment (aabadie) [inline tests/unittests/tests-core/tests-core-byteorder.c:139]: You forgot the function names here ;)

File: tests/unittests/tests-ieee802154/tests-ieee802154.c (hunk 2 above)
Comment (aabadie) [reply tests/unittests/tests-ieee802154/tests-ieee802154.c:30]: It's in its own commit which explains why. So ok!

File: tests/unittests/tests-core/tests-core-byteorder.c (hunk 3 above)
Comment (benpicco) [reply tests/unittests/tests-core/tests-core-byteorder.c]: updated.

**Comment by aabadie** [comment]
Please squash!

**Review by aabadie — APPROVED** [review]
ACK
